it/v1/endpoints/CreateAndAmendCharitableGivingControllerISpec.scala

```diff
@@ -0,0 +1,246 @@
+        response.status shouldBe OK
+        response.json shouldBe responseBody
+        response.header("X-CorrelationId").nonEmpty shouldBe true
+      }
+    }
+
+    "return error according to spec" when {
+
+      "validation error" when {
+        s"an invalid NINO is provided" in new Test {
+          override val nino: String = "INVALID_NINO"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestJson))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(NinoFormatError)
+        }
+        s"an invalid taxYear is provided" in new Test {
+          override val mtdTaxYear: String = "INVALID_TAXYEAR"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestJson))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(TaxYearFormatError)
+        }
+        s"an invalid /giftAidPayments/totalAmount is provided" in new Test {
+          override val requestJson: JsValue = Json.parse(
+            s"""
+               |{
+               |  "giftAidPayments": {
+               |    "totalAmount": -1
+               |  }
+               |}
+               |""".stripMargin
+          )
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestJson))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(ValueFormatError.copy(paths = Some(Seq("/giftAidPayments/totalAmount"))))
+        }
+        s"a taxYear with range of greater than a year is provided" in new Test {
+          override val mtdTaxYear: String = "2019-21"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestJson))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(RuleTaxYearRangeInvalidError)
+        }
+
+        s"a taxYear below 2019-21 is provided" in new Test {
+          override val mtdTaxYear: String = "2017-18"
+          override val desTaxYear: String = "2018"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+            DesStub.onError(DesStub.POST, desUri(desTaxYear), BAD_REQUEST, errorBody("INVALID_ACCOUNTING_PERIOD"))
+          }
+
```

Comment:
The test name doesn't match what is being tested.

The test should be testing that when the `taxYear` provided is before 2017-18 (e.g. 2016-17) then it should give the `RULE_TAX_YEAR_NOT_SUPPORTED` Error  with the des stub response not being altered